### PR TITLE
[entraid] store entra `appId` in plugin settings

### PR DESCRIPTION
**Pull request by tigrato**
This PR stores the Entra ID appID of the application used for SSO in Entra ID plugin settings. This field filled in a best effort scenario and might be empty for old Entra Plugins. Newly created plugins will make the field mandatory.

### CODE REVIEW

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -6498,6 +6498,11 @@ message PluginEntraIDSyncSettings {
   // This field is populated on a best-effort basis for legacy plugins but mandatory for plugins created after its introduction.
   // For existing plugins, it is filled in using the Entra integration when utilized.
   string tenant_id = 4;
+
+  // app_id refers to the Azure Application ID that supports the SSO for "sso_connector_id".
+  // This field is populated on a best-effort basis for legacy plugins but mandatory for plugins created after its introduction.
+  // For existing plugins, it is filled in using the entity descriptor url when utilized.
+  string app_id = 5;
```
Comment (codingllama) [inline api/proto/teleport/legacy/types/types.proto:6505]: WDYT about azure_app_id ?

File: lib/msgraph/models.go [hunk 2]
```diff
@@ -74,6 +74,24 @@ type Application struct {
 	OptionalClaims        *OptionalClaims `json:"optionalClaims,omitempty"`
 }
 
+const (
+	// OPTIONAL_CLAIM_GROUP_NAME is the group claim.
+	OPTIONAL_CLAIM_GROUP_NAME = "groups"
+)
+
+const (
+	// OPTIONAL_CLAIM_ADDITIONAL_PROPERTIES_SAM_ACCOUNT_NAME is the sAMAccountName claim.
```
Comment (codingllama) [inline lib/msgraph/models.go:83]: Is "sAMAccountName" correct? Should it be SAMAccountName?

Same for other lines.

**Review by codingllama — COMMENTED** [review]
Looks good.

File: lib/msgraph/models.go (hunk 2 above)
Comment (tigrato) [reply lib/msgraph/models.go:83]: yes, https://learn.microsoft.com/en-us/entra/identity/hybrid/connect/how-to-connect-fed-group-claims#group-claims-for-applications-migrating-from-ad-fs-and-other-identity-providers

File: api/proto/teleport/legacy/types/types.proto (hunk 1 above)
Comment (tigrato) [reply api/proto/teleport/legacy/types/types.proto:6505]: renamed to entra_app_id

File: lib/msgraph/models.go (hunk 2 above)
Comment (codingllama) [reply lib/msgraph/models.go:83]: Weird 🤷‍♂️. Thanks for the link.

**Review by codingllama — APPROVED** [review]

File: api/proto/teleport/legacy/types/types.proto [hunk 3]
```diff
@@ -6498,6 +6498,11 @@ message PluginEntraIDSyncSettings {
   // This field is populated on a best-effort basis for legacy plugins but mandatory for plugins created after its introduction.
   // For existing plugins, it is filled in using the Entra integration when utilized.
   string tenant_id = 4;
+
+  // entra_app_id refers to the Azure Application ID that supports the SSO for "sso_connector_id".
```
Comment (flyinghermit) [inline api/proto/teleport/legacy/types/types.proto:6502]: nit: should "Azure Application ID" be "Entra Application ID"?

File: lib/msgraph/models.go [hunk 4]
```diff
@@ -83,10 +101,10 @@ type OptionalClaim struct {
...
 	// - emit_as_roles
 	// - cloud_displayname
-	AdditionalProperties *[]string `json:"additionalProperties,omitempty"`
-	Essential            *bool     `json:"essential,omitempty"`
-	Name                 *string   `json:"name,omitempty"`
-	Source               *string   `json:"source,omitempty"`
+	AdditionalProperties []string `json:"additionalProperties,omitempty"`
+	Essential            *bool    `json:"essential,omitempty"`
+	Name                 *string  `json:"name,omitempty"`
+	Source               *string  `json:"source,omitempty"`
```
Comment (flyinghermit) [inline lib/msgraph/models.go:107]: nit: Godoc would be helpful here to understand what kind of data/value are we expecting for each field. For example, what could be the possible value for `Source`?

**Review by flyinghermit — APPROVED** [review]

**Pull request merged**